### ISSUE TITLE
Unable to setup Prometheus Stack

### ISSUE THREAD

**Issue by hakuna-matatah** [kubernetes/perf-tests#2464]
**What happened**:
I think since the merge of this [PR](https://github.com/kubernetes/perf-tests/pull/2421) , Prometheus stack is failing to setup.
It appears that template changes and `0prometheus-operator-deployment.yaml` file was changed in the [PR here](https://github.com/kubernetes/perf-tests/pull/2421/files#diff-e3787fcec46425dd9676d34f8196aa4c547c0059239736a68bee80863fa4280bR63) 

```
I1215 21:09:39.405672      22 prometheus.go:231] Setting up prometheus stack
I1215 21:09:39.432486      22 framework.go:276] Applying templates for "*.yaml"
I1215 21:09:39.432579      22 framework.go:287] Applying 0prometheus-operator-0alertmanagerConfigCustomResourceDefinition.yaml
I1215 21:09:39.497819      22 framework.go:287] Applying 0prometheus-operator-0alertmanagerCustomResourceDefinition.yaml
I1215 21:09:39.636420      22 framework.go:287] Applying 0prometheus-operator-0podmonitorCustomResourceDefinition.yaml
I1215 21:09:39.653045      22 framework.go:287] Applying 0prometheus-operator-0probeCustomResourceDefinition.yaml
I1215 21:09:39.668810      22 framework.go:287] Applying 0prometheus-operator-0prometheusCustomResourceDefinition.yaml
I1215 21:09:39.794738      22 framework.go:287] Applying 0prometheus-operator-0prometheusruleCustomResourceDefinition.yaml
I1215 21:09:39.804634      22 framework.go:287] Applying 0prometheus-operator-0servicemonitorCustomResourceDefinition.yaml
I1215 21:09:39.824190      22 framework.go:287] Applying 0prometheus-operator-0thanosrulerCustomResourceDefinition.yaml
I1215 21:09:39.926123      22 framework.go:287] Applying 0prometheus-operator-clusterRole.yaml
I1215 21:09:39.935629      22 framework.go:287] Applying 0prometheus-operator-clusterRoleBinding.yaml
I1215 21:09:39.943881      22 framework.go:287] Applying 0prometheus-operator-deployment.yaml
F1215 21:09:39.944288      22 clusterloader.go:326] Error while setting up prometheus stack: TemplateToObject error: replacing placeholders error: Invalid YAML Template
0
````

**What you expected to happen**:
Prometheus stack should be able to setup cleanly without issues.

**How to reproduce it (as minimally and precisely as possible)**:

Run CL2 loadloadtest with Prometheus stack enabled.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

**Comment by jprzychodzen** [comment]
@hakuna-matatah Could you point me to a specific test that is failing?

Presubmit which run on perf-tests repository has passed, and those tests create Prometheus stack (see https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/perf-tests/2421/pull-perf-tests-clusterloader2/1734128255960616960 for a presubmit tests for #2421).

**Comment by hakuna-matatah** [comment]
>Could you point me to a specific test that is failing?

Above logs are coming from the test we run on EKS internally, and we do run at 5k nodes scale. So, may be at 5k scale, some  `vars` are not getting set properly after this change ?
And i only started seeing this issue since this template changes were merged and when I reverted those changes locally to test the hypothesis, it started working fine, i.e; it went ahead and installed prometheus stack fine.

**Comment by hakuna-matatah** [comment]
Also, @jprzychodzen it appears that above test is also not setting the `PROMETHEUS_NODE_SELECTOR` ? Do you have an example of any test where it sets `PROMETHEUS_NODE_SELECTOR` this flag and works fine ?

**Comment by jprzychodzen** [comment]
@hakuna-matatah , we don't have any CI in OSS which uses this flag, so it went under the radar.

Fix is in review in #2482
